route.go

```diff
@@ -20,14 +20,15 @@ type Route[ResponseBody any, RequestBody any] struct {
 	BaseRoute
 }
 
-func NewBaseRoute(method, path string, handler any, openapi *OpenAPI, options ...func(*BaseRoute)) BaseRoute {
+func NewBaseRoute(method, path string, handler any, e *Engine, options ...func(*BaseRoute)) BaseRoute {
```

Comment:
But do we really need this? For future compatibility maybe?

Reply:
I figure we can just deprecate this function. You are just talking about the function right? Deprecating the Struct of BaseRoute would be a fairly large change which we'd probably want a separate PR for.

Reply:
No, I was talking about changing the *OpenAPI param to *engine

Reply:
Oh. It fixed the WithContentTypes option. I think the line you're missing is:

https://github.com/go-fuego/fuego/pull/334/files#diff-6b1e829598248403d0c671996c16c10f8a2d8db33f96b6db158654b78103dec4R31

see the `AcceptedContentTypes: e.acceptedContentTypes,`

Reply:
Hence my 4 questions where do we want to put `acceptedContentTypes` and how do we want to provide an option to configure it 😄

Reply:
Let me know if that clarifies things. I personally don't mind `acceptedContentTypes` existing on the Engine. At the very least I want avoid having to support a set of options for NewOpenAPI as we're getting to be pretty nested at that point. 

So my preference is: 1 or 2

Reply:
Hey @EwenQuim any more word on this?